### PR TITLE
spark-model: default lossless single-warp GEMV on decode

### PR DESCRIPTION
**Pull request by facepunch47**
Decode on GB10 is memory-bandwidth bound. The bit-identical single-warp GEMV (`w4a16_gemv_sw`, 8 outputs/block, no cross-warp smem reduce) already exists and the microtest requires 100% bit-identity — but it was opt-in behind `ATLAS_DECODE_OPT` and the default split-SiLU **down** projection never used it at all.

This ships that kernel on the decode path that actually runs.

## What changes

- **Default ON** via `ModelLevers::gemv_sw`. Kill switch: `ATLAS_NO_GEMV_SW=1` (exactly `"1"`; `=0` does not disable).
- Dense FFN: `w4a16_gemv_dual_sw` for gate+up; `w4a16_gemv_sw` for split-SiLU / GELU down. Dual and silu-input SW are independent — a missing `silu_input_sw` no longer skips `dual_sw`.
- SSM sequential QKVZ + out_proj: same `w4a16_decode_gemv` helper.
- Grid SSOT: Rust `W4A16_GEMV_SW_OUTS_PER_BLOCK = 8` is pinned to CUDA `#define N_PER_BLOCK_SW 8`. Swapping the kernel without swapping the grid writes the wrong outputs.

Out of scope (follow-up): attention / MTP / Nemotron `w4a16_gemv` call sites — same helper, more wiring.

## Why this is a product win

```mermaid
flowchart LR
  subgraph bound [GB10 decode]
    BW["273 GB/s LPDDR5X"]
    GEMV["W4A16 GEMV"]
    GEMV --> BW
  end
  subgraph before [Before]
    B1["dual GEMV 64-thread"]
    B2["silu_mul"]
    B3["down GEMV 64-thread"]
    B1 --> B2 --> B3
  end
  subgraph after [After — default]
    A1["dual_sw 8 outs/block"]
    A2["silu_mul"]
    A3["gemv_sw 8 outs/block"]
    A1 --> A2 --> A3
  end
  before --> after
```

Same bytes moved, fewer idle warps waiting on `__syncthreads`, same numerics. The kernel was already in `kernels/gb10/common/w4a16_gemv.cu`; this is the dispatch default.

## Tests

- `gemv_sw_from` polarity (`unset`/`=0` ON, `=1` OFF)
- lever + live handle required
- SW grid covers N and is half the base grid when N is divisible by 8
- CUDA `#define N_PER_BLOCK{,_SW}` matches the Rust constants on every `w4a16_gemv.cu` / `_fused.cu` copy
- `ModelLevers::defaults().gemv_sw` is ON

GPU oracle (not CI): `cargo run --release -p spark-model --example w4a16_gemv_sw_microtest`

## Related

Persisting slot-keyed decode graphs (fewer recaptures) is https://github.com/Avarok-Cybersecurity/atlas/pull/478 — complementary, not overlapping.

## GB10 serve proof

- Bit-identity: `w4a16_gemv_sw_microtest` on this GB10, seed `0x51A7`, 100% match vs 64-thread `w4a16_gemv`.
- Kernel SSOT: `kernels/gb10/common/w4a16_gemv.cu` — `w4a16_gemv` / `_sw` share `w4a16_gemv_partial`; dual/dual_sw share `w4a16_dual_partial`. strix/strix-hip `.cu` are symlinks. Grid: base `ceil(N/4)`, SW `ceil(N/8)`, block 256.
- Rebased onto `origin/main` (`680b3a5`, includes #473 TUI). Independent of #478/#481.
- Kill switch is `ATLAS_NO_GEMV_SW=1` only (`=0` does not disable).
- Out of scope: main-model `w4a16_gemv_logits`; attention/MTP/Nemotron wiring is #480.

## CLA

- [x] I have read and agree to the [Contributor License Agreement](../CLA.md).

### CODE REVIEW

**Comment by facepunch47** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by facepunch47** [comment]
recheck

**Comment by tbraun96** [comment]
Merged into main via Batch 5 (#490), landed as `7761b9653`. Squash-merging a stack does not auto-close its members, so closing manually — merged, not abandoned.
